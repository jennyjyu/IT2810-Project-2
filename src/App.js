import React, {Component} from 'react';
import Header from './components/Header';
import Tabs from './components/Tabs'
import './css/main.css';
// import { render } from "react-dom";
import Footer from "./components/Footer"
import MediaCategory from "./components/MediaCategory"
import catData from "./categoryData"


class App extends Component {
    constructor(props) {
        super(props)
        this.state = {
            data: catData,
            selectedButton: {
                image: "Animals",
                sound: "Cartoons",
                text: "Animals"
            },
            selected: [0, 0, 0],
            hasFavorite: localStorage.getItem('hasFavorite'),
            clicks: Number(sessionStorage.getItem('clicks'))
        }
    }

    incrementClick = (prevState) => {
        sessionStorage.setItem('clicks', this.state.clicks + 1)
        this.setState({
            clicks: this.state.clicks + 1
        })
    }

    handleRadioChange = (button, i) => {
        const sel = [...[], ...this.state.selected];
        if (button.mediaLabel === "Images") {
            sel[0] = i;
            this.setState( prevState => ({
                selectedButton: {
                    sound: prevState.selectedButton.sound,
                    image: button.categories[i].text,
                    text: prevState.selectedButton.text
                },
                selected: sel,
            }))
        }
        if (button.mediaLabel === "Sounds") {
            sel[1] = i;
            this.setState(prevState => ({
                selectedButton: {
                    sound: button.categories[i].text,
                    image: prevState.selectedButton.image,
                    text: prevState.selectedButton.text
                },
                selected: sel,
            }))
        }
        if (button.mediaLabel === "Texts") {
            sel[2] = i;
            this.setState(prevState => ({
                selectedButton: {
                    text: button.categories[i].text,
                    image: prevState.selectedButton.image,
                    sound: prevState.selectedButton.sound
                },
                selected: sel,
            }))
        }
    }

    handleClick = () => {
        localStorage.setItem('favoriteImage', this.state.selectedButton.image)
        localStorage.setItem('favoriteSound', this.state.selectedButton.sound)
        localStorage.setItem('favoriteText', this.state.selectedButton.text)
        localStorage.setItem('hasFavorite', true)
        this.setState({
            hasFavorite: true
        })
    }

    applyFavorite = () => {
        this.setState({
            selectedButton: {
                image: localStorage.getItem('favoriteImage'),
                text: localStorage.getItem('favoriteText'),
                sound: localStorage.getItem('favoriteSound')
            },
            selected: [
                this.state.data[0].categories.findIndex((el) => el.text === localStorage.getItem('favoriteImage')),
                this.state.data[1].categories.findIndex((el) => el.text === localStorage.getItem('favoriteSound')),
                this.state.data[2].categories.findIndex((el) => el.text === localStorage.getItem('favoriteText'))
            ]
        })
    }


    handleRemove = () => {
        this.setState({
            hasFavorite: false
        })
        localStorage.clear()
    }

    render() {
        const mediaCategories = this.state.data.map((data,i) =>
            <MediaCategory
                mediaLabel={data.mediaLabel}
                key={data.id}
                categories={data.categories}
                handleRadioChange={this.handleRadioChange}
                index={this.state.selected[i]}
            />);
        return (
            <div onClick={this.incrementClick}>
                <Header headerText="Project 2"/>
                <div className="main-content">

                    <Tabs
                        mediaCategories={this.state.data}
                        selectedButton={this.state.selectedButton}
                        hasFavorite={this.state.hasFavorite}
                    >
                        <div label="1"/>
                        <div label="2"/>
                        <div label="3"/>
                        <div label="4"/>
                    </Tabs>
                    <div className="media-categories">
                        <div className="categories">{mediaCategories}</div>

                        <div className="buttons">
                            {this.state.hasFavorite
                                ? <button onClick={this.handleRemove}>Remove favorite</button>
                                : <button onClick={this.handleClick}>Add to favorite</button>
                            }
                            {this.state.hasFavorite
                                ? <button onClick={this.applyFavorite}>Apply favorite</button>
                                : null
                            }
                            <p>Your clicks: {this.state.clicks}</p>


                        </div>
                    </div>
                </div>
                <Footer />
            </div>
        );
    }

}

export default App;
