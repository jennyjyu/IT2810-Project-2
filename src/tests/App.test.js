import React from 'react';
import App from '../App';
import renderer from 'react-test-renderer'


test("App snapshot test", () => {
  const tree = renderer.create(
     <App />
  )

  expect(tree).toMatchSnapshot()
})
