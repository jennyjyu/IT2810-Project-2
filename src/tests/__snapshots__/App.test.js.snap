// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot test 1`] = `
<div
  onClick={[Function]}
>
  <div
    className="header-container"
  >
    <header
      className="header"
    >
      <h1>
        Project 2
      </h1>
    </header>
  </div>
  <div
    className="main-content"
  >
    <div
      className="tabs"
    >
      <ol
        className="tab-list"
      >
        <li
          className="tab-list-item tab-list-active"
          onClick={[Function]}
        >
          Tab 
          1
        </li>
        <li
          className="tab-list-item"
          onClick={[Function]}
        >
          Tab 
          2
        </li>
        <li
          className="tab-list-item"
          onClick={[Function]}
        >
          Tab 
          3
        </li>
        <li
          className="tab-list-item"
          onClick={[Function]}
        >
          Tab 
          4
        </li>
      </ol>
      <div
        className="tab-content"
      />
      <div
        label="1"
      >
        <div
          className="tab-content-container"
        >
          <div
            className="image"
          >
            <div
              dangerouslySetInnerHTML={
                Object {
                  "__html": "",
                }
              }
              id="svg_img"
              style={
                Object {
                  "marginLeft": "50px",
                  "marginTop": "0px",
                }
              }
            />
          </div>
          <div
            className="nested"
          >
            <div
              className="text-wrap"
            >
              <p>
                
              </p>
              <p>
                 - 
                
              </p>
            </div>
            <div
              className="audio-wrap"
            >
              <audio
                id="Player"
                src="./media/sounds/Cartoons/1.mp3"
              />
              <button
                className="controls"
                onClick={[Function]}
              >
                â–º / 
                <b>
                  ||
                </b>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="media-categories"
    >
      <div
        className="categories"
      >
        <div
          className="category-container"
        >
          <label>
            Images
            <ul
              className="media-list"
            >
              <label
                className="radio-label"
              >
                <input
                  checked={true}
                  defaultChecked={true}
                  name="Images"
                  onClick={[Function]}
                  type="radio"
                />
                Animals
                <br />
              </label>
              <label
                className="radio-label"
              >
                <input
                  checked={false}
                  defaultChecked={false}
                  name="Images"
                  onClick={[Function]}
                  type="radio"
                />
                Vehicles
                <br />
              </label>
              <label
                className="radio-label"
              >
                <input
                  checked={false}
                  defaultChecked={false}
                  name="Images"
                  onClick={[Function]}
                  type="radio"
                />
                Nature
                <br />
              </label>
            </ul>
          </label>
        </div>
        <div
          className="category-container"
        >
          <label>
            Sounds
            <ul
              className="media-list"
            >
              <label
                className="radio-label"
              >
                <input
                  checked={true}
                  defaultChecked={true}
                  name="Sounds"
                  onClick={[Function]}
                  type="radio"
                />
                Cartoons
                <br />
              </label>
              <label
                className="radio-label"
              >
                <input
                  checked={false}
                  defaultChecked={false}
                  name="Sounds"
                  onClick={[Function]}
                  type="radio"
                />
                Jazz
                <br />
              </label>
              <label
                className="radio-label"
              >
                <input
                  checked={false}
                  defaultChecked={false}
                  name="Sounds"
                  onClick={[Function]}
                  type="radio"
                />
                Horror
                <br />
              </label>
            </ul>
          </label>
        </div>
        <div
          className="category-container"
        >
          <label>
            Texts
            <ul
              className="media-list"
            >
              <label
                className="radio-label"
              >
                <input
                  checked={true}
                  defaultChecked={true}
                  name="Texts"
                  onClick={[Function]}
                  type="radio"
                />
                Animals
                <br />
              </label>
              <label
                className="radio-label"
              >
                <input
                  checked={false}
                  defaultChecked={false}
                  name="Texts"
                  onClick={[Function]}
                  type="radio"
                />
                Creepy
                <br />
              </label>
              <label
                className="radio-label"
              >
                <input
                  checked={false}
                  defaultChecked={false}
                  name="Texts"
                  onClick={[Function]}
                  type="radio"
                />
                Jokes
                <br />
              </label>
            </ul>
          </label>
        </div>
      </div>
      <div
        className="buttons"
      >
        <button
          onClick={[Function]}
        >
          Add to favorite
        </button>
        <p>
          Your clicks: 
          0
        </p>
      </div>
    </div>
  </div>
  <div
    className="footer-container"
  >
    <footer />
  </div>
</div>
`;
